### PR TITLE
feat: add grpc health service

### LINKED ISSUE #378: Support a health check endpoint to use as readiness and liveness kubernetes probes
**Is your feature request related to a problem? Please describe.**
Lack of readiness / liveness probes, which is needed for a graceful server shutdown.

**Describe the solution you'd like**
I'd like to get a gRPC server for health check, and use it to set up the readiness and liveness probes when running in kubernetes.

**Describe alternatives you've considered**
Not sure if there is a good alternative.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by hongalex**
**What type of PR is this?**
> /kind feat

**What this PR does / Why we need it**:
Adds gRPC health check service to the grpc server.

**Which issue(s) this PR fixes**:
Closes #378

**Special notes for your reviewer**:
Replaces part of #381

### CODE REVIEW

**Comment by yuryu** [comment]
overall lgtm. Do you think it makes sense to add a simple test like

```go
c := healthgrpc.NewHealthClient(conn)
got, err := c.Check(ctx, request)
if err != nil {
 // error
}
if got is not what we want {
  // error
}
```

**Review by yuryu — APPROVED** [review]

File: internal/app/server/open_saves_test.go [hunk 1]
```diff
@@ -56,6 +62,45 @@ const (
...
+		t.Fatalf("did not connect %v", err)
+	}
+	defer conn.Close()
+
+	hc := healthgrpc.NewHealthClient(conn)
+	got, err := hc.Check(ctx, &healthgrpc.HealthCheckRequest{
+		Service: serviceName,
+	})
+	if err != nil {
+		t.Fatalf("healthClient.Check err: %v", err)
```
Comment (yuryu) [inline internal/app/server/open_saves_test.go:97]: suggest Errorf instead 

https://github.com/golang/go/wiki/TestComments#keep-going

File: internal/app/server/open_saves_test.go [hunk 2]
```diff
@@ -56,6 +62,45 @@ const (
...
+	}
+	if want := healthgrpc.HealthCheckResponse_SERVING; got.Status != want {
+		t.Fatalf("hc.Check got: %v, want: %v", got.Status, want)
```
Comment (yuryu) [inline internal/app/server/open_saves_test.go:100]: ditto

**Review by yuryu — APPROVED** [review]
lgtm.

This is more like a wishlist rather than a request but I guess other tests should also use `Run()` and would also want to change `Run()` to accept a `Listener` instead of direct network address/port so that this test can use `bufconn` instead.

**Comment by hongalex** [comment]
> This is more like a wishlist rather than a request but I guess other tests should also use `Run()` and would also want to change `Run()` to accept a `Listener` instead of direct network address/port so that this test can use `bufconn` instead.

Agreed. I'll open a issue for this.

**Pull request merged**